x/stake/handler.go

```diff
@@ -130,13 +130,13 @@ func handleMsgEditValidator(ctx sdk.Context, msg types.MsgEditValidator, k keepe
 	validator.Description = description
 
 	if msg.CommissionRate != nil {
-		if err := k.UpdateValidatorCommission(ctx, validator, *msg.CommissionRate); err != nil {
+		commission, err := k.UpdateValidatorCommission(ctx, validator, *msg.CommissionRate)
```

Comment:
why is this using a pointer dereference here?

Reply:
I think you wrote it - per a comment in `x/stake/types/msg.go`, we use a pointer so that updating the commission with `MsgEditValidator` is optional.

Reply:
If you're asking why `if msg.CommissionRate != nil { // ... }`, it's due to what @cwgoes stated. In `MsgEditValidator`, the commission is optional and because nil decimals don't deserialize correctly, I had to make it a pointer.

ref: #2373

Once that is addressed, we can remove the pointer reference and just check `msg.CommissionRate.IsNil()`.